src/lib.rs

```diff
@@ -700,13 +699,17 @@ impl GenerateTasksService for GenerateTasksServiceImpl {
                 };
 
                 if let Some(generated_task) = generated_tasks.get(&task_name) {
+                    let large_distro = self
```

Comment:
[nit] I think this variable naming is a bit odd. This could just the default distro configured for this variant instead right?

Reply:
Not really. The function name was misleading. We are trying to determine exactly large distro name when we have to run the task on the large distro. We explicitly set large distro to run the task on when needed. When we don't need to run it on the large distro we do nothing, so it will run on a normal distro by default.

I've renamed the function to reflect that.